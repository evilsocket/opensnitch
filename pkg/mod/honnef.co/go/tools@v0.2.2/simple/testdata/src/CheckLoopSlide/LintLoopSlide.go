package pkg

func fn() {
	var n int
	var bs []int
	var offset int

	for i := 0; i < n; i++ { // want `should use copy\(\) instead of loop for sliding slice elements`
		bs[i] = bs[offset+i]
	}

	for i := 1; i < n; i++ { // not currently supported
		bs[i] = bs[offset+i]
	}

	for i := 1; i < n; i++ { // not currently supported
		bs[i] = bs[i+offset]
	}

	for i := 0; i <= n; i++ {
		bs[i] = bs[offset+i]
	}
}
