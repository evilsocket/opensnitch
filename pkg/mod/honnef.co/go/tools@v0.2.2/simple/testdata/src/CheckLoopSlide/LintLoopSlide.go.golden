package pkg

func fn() {
	var n int
	var bs []int
	var offset int

	copy(bs[:n], bs[offset:])

	for i := 1; i < n; i++ { // not currently supported
		bs[i] = bs[offset+i]
	}

	for i := 1; i < n; i++ { // not currently supported
		bs[i] = bs[i+offset]
	}

	for i := 0; i <= n; i++ {
		bs[i] = bs[offset+i]
	}
}
